### PR TITLE
Common configuration utilities for Swift's HTTP services.

### PR DESCRIPTION
**Pull request by mookerji**
- Provides a default configuration type and some generic behavior for
  populating them from command line options or environmental
  parameters.
- Ports tests from the broker project.
- Moved Conf => Config.

/cc @mfine @JoshuaGross

### CODE REVIEW

**Comment by mfine** [comment]
This looks good! Put comments in https://github.com/mookerji/skylark-core/pull/1. Mostly just formatting things.

Still not fully understanding how merge works - its combining the fields from the two parameters? What does it mean by "second non-Nothing Value"? I should go look at the Data.Monoid definition...

Excited to pull this into broker, api, and relay!

**Comment by JoshuaGross** [comment]
Looks good. This just covers 4 configs common to all Skylark projects right now, right? ```+    makeEnv [ "SKYLARK_CONFFILE" .= _cConfigFile
+            , "SKYLARK_PORT"     .= _cPort
+            , "SKYLARK_TIMEOUT"  .= _cTimeout
+            , "SKYLARK_LOGLEVEL" .= _cLogLevel```

so each project still needs to manage its own configs separately?

**Comment by mookerji** [comment]
@JoshuaGross yep.

File: src/Network/Skylark/Core/Conf.hs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                                        -- configuration file.
+                -> IO (Either String a) -- ^ Either an error string
+                                        -- or configuration.
+getCompleteConf p conf = do
+  e <- decodeEnv :: IO (Either String a)
+  let rest x = do opt <- options p
+                  d <- def
+                  f <- getDataFile $ fromMaybe "" $ conf (d <> opt <> x)
+                  return $ Right (d <> f <> opt <> x)
+  either (return . Left) rest e
```
Comment (mfine) [inline src/Network/Skylark/Core/Conf.hs]: What about passing in the results of `execParser` and `getDataFile` (and the env?) here vs calling them repeatedly for each field? There should be a way to formulate this without having to continuously reparse everything.

Comment (mfine) [inline src/Network/Skylark/Core/Conf.hs]: Oh, is this grabbing everything? Hmmm - yeah looks like it - thought it was getting called per item. nm.

**Pull request merged**